### ISSUE TITLE
Hunger Games tells there are no questions left (when there actually is)

### ISSUE THREAD

**Issue by alexgarel** [openfoodfacts/openfoodfacts-hungergames#394]
### What
- Hunger Games tells there are no questions left (when there actually is)
- Playing with https://hunger.openfoodfacts.org/questions?value_tag=en%3Ateas&type=category
- After some time, I end up with :

![Capture d’écran de 2022-02-28 18-44-33](https://user-images.githubusercontent.com/144788/156032778-19f84d2c-60f7-459e-9dbd-6e689d550761.png)

- It says there is no more questions while there are 7046 remaining. If I refresh the page, it gives me examples (but I have the one I ignored again…)
- It seems easy to reproduce if you always click on "ignore"

**Comment by alexfauquette** [comment]
Thanks for reporting :)

The problem seems to be the following one: 

- We verify if the server send the correct number of insights. If we get less than requested, we assume it's because we are at the end of the batch
https://github.com/alexfauquette/openfoodfacts-hungergames-1/blob/23ca2a0797e914497f04ef360f5e0a48a7944f46/src/views/QuestionView.vue#L238-L243

- But we filter insights that do not have source image (no image to display with the question)
https://github.com/alexfauquette/openfoodfacts-hungergames-1/blob/23ca2a0797e914497f04ef360f5e0a48a7944f46/src/robotoff.js#L26-L30

So we move the filtering after adding the `NO_QUESTION_LEFT` or rely on another information to add the `NO_QUESTION_LEFT`

**Comment by sumana2001** [comment]
Hey @alexfauquette @alexgarel! I am interested to work on this. 
Meanwhile my PR #399 is in review, I was planning to contribute to this issue. Would be great if you could assign it to me.

**Comment by sumana2001** [comment]
Hey @alexfauquette! I have been trying this issue for so long but finally understood what the problem is, but I am not able to understand how to fix it. Could you please guide me?

So the issue is that [here](https://github.com/openfoodfacts/openfoodfacts-hungergames/blob/ea920a7f4ff34a85c3c506059b10118a3bf4110c/src/robotoff.js#L38-L51) in this get request, we are calling the first 10 records in database according to some sorting parameters. Then we are checking if that particular record has already been visited, and we add only the records which have not be visited to dataToAdd array [here](https://github.com/openfoodfacts/openfoodfacts-hungergames/blob/ea920a7f4ff34a85c3c506059b10118a3bf4110c/src/views/QuestionView.vue#L233-L239). But since everytime we call [loadQuestions](https://github.com/openfoodfacts/openfoodfacts-hungergames/blob/ea920a7f4ff34a85c3c506059b10118a3bf4110c/src/views/QuestionView.vue#L200-L252) function, the same 10 records are being returned, no new question is able to get into the dataToAdd array, thus resulting in No Questions Left.

I explored a lot of possibilities but I think this can only be fixed if we have a separate function for it in the backend which keeps track of till where the records have been returned, sort of like pagination. Could you tell me how I can get the code of backend? Or if there is any other way of approaching this problem?

**Comment by alexfauquette** [comment]
Yes, that's not possible to fix. My bad, I observed that some data was removed because they do not have images. But I did not realized that next call will return same data

I made a [PR](https://github.com/openfoodfacts/robotoff/pull/658) to add pagination, such that we will be able to get more data

**Comment by sumana2001** [comment]
Okay great! Would continue working on it once the pagination PR gets merged.

**Comment by sumana2001** [comment]
Hey @alexfauquette
Since the PR for pagination has been merged, should I start working on this issue again or is it still on hold?

**Comment by alexgarel** [comment]
@sumana2001 I will deploy in prod on monday, you may test using robotoff.openfoodfacts.net, but there are not many question (you can edit some products on openfoodfacts.net to trigger questions if you need)

**Comment by alexfauquette** [comment]
@sumana2001 The easiest way is to wait for the Monday release. You can already think about how to modify the code to use pagination

**Comment by sumana2001** [comment]
Alright! I'll start on Monday. Thanks

**Comment by alexfauquette** [comment]
@sumana2001 robotoff has been released. Now random is random and sorted product support pagination by using `page` and `count` query parameters

**Comment by sumana2001** [comment]
Okay great! I'll start working on it.

**Comment by sumana2001** [comment]
Hi @alexfauquette and @alexgarel, I am facing some issues in understanding how pagination is working. I tried a lot of console logs and finally started testing on Postman. As mentioned above I passed count and page as query parameters. Although the count is working fine on changing its values, the page parameter gives the same response for whatever page number I give. Could you please guide me on where I am going wrong?

Putting count=5, page=1
<img width="1060" alt="Screenshot 2022-04-05 at 7 40 46 PM" src="https://user-images.githubusercontent.com/63084088/161773797-1fdcf4b3-010c-4ef2-abeb-699bd562e921.png">
Putting count=5, page=4
<img width="1060" alt="Screenshot 2022-04-05 at 7 40 34 PM" src="https://user-images.githubusercontent.com/63084088/161773760-2e47391a-8998-4475-b0ae-4d7a5f03b1a4.png">

**Comment by alexgarel** [comment]
@sumana2001 I think we broke it in last release !
Time to add a test !

**Comment by alexgarel** [comment]
@sumana2001 it should be ok now !

**Comment by sumana2001** [comment]
Thanks @alexgarel for the quick fix! Will try to fix this issue asap
